isos/appliance.sh

```diff
@@ -126,6 +138,39 @@ cp ${BIN}/vic-init $(rootfs_dir $PKGDIR)/sbin/vic-init
 cp ${BIN}/{docker-engine-server,port-layer-server,vicadmin} $(rootfs_dir $PKGDIR)/sbin/
 cp ${BIN}/unpack $(rootfs_dir $PKGDIR)/bin/
 
+APPLIANCE_NAME=appliance.iso
+
+if [ -n "${OUTPUTFILE}" ]; then
+	APPLIANCE_NAME=${OUTPUTFILE}
+else
+	APPLIANCE_NAME=appliance.iso
+fi
+
+echo Appliance name: $APPLIANCE_NAME
+
+if [ -n "${XTRABINS}" ]; then
+    for XBIN in $XTRABINS; do
+        GS=$(echo ${XBIN} | grep '^gs://' | cat)
+        if [ -n "$GS" ]; then
+            XBIN_LATEST_BUILD="$(gsutil ls -l ${XBIN} | grep -v TOTAL | sort -k2 -r | (trap ' ' PIPE; head -1))"
```

Comment:
unless this is expected to change build by build it should be in staging along with all the other remote package dependencies.

Honestly it looks like you want to have a separate (or additional) staging step, and a separate final build step instead of trying to inline it into the current appliance.sh script.